aiida/orm/nodes/process/process.py

```diff
@@ -479,7 +479,15 @@ def is_valid_cache(self):
 
         :returns: True if this process node is valid to be used for caching, False otherwise
         """
-        return super().is_valid_cache and self.is_finished
+        if not (super().is_valid_cache and self.is_finished):
+            return False
```

Comment:
why not
```
if super().is_valid_cache and self.is_finished:
    return True
```

Reply:
That's not the same logic -- in the case where the two statements are true, we want to _also_ perform the checks implemented after that.

Reply:
good point